### ISSUE TITLE
Feature Request: Arbitrary Tags

### ISSUE THREAD

**Issue by Haaruun-I** [knuxify/eartag#30]
The main reason for this is the large amount of niche tags out there, like composer which is commonly used for audiobooks.
Instead of adding a new field for every new tag, you could have a way to add an arbitrary field, and show an error on saving if the lib rejects it.

Here's a quick mockup to get across my idea
![The eartag file editor view, with the last item having 2 input boxes, one where the name of the tag would be and the other for the value. There is a plus button next to the field ](https://user-images.githubusercontent.com/22063464/190573336-e1fc1b33-2f05-4c4d-b3ee-cf79889a8c3b.png)

 It would be my first time doing gnome development, but id be willing to try implementing it.

**Comment by knuxify** [comment]
This was originally planned to be added, but was indefinitely postponed. Interesting to see that there's a use case for it!

The main issue with adding something like this is that every file tagging format has support for different tags. While there's overlap between most file formats, some more exotic tags may only be implemented in one format.

This doesn't mean that this will be impossible; one way I see it being possible is to allow each file type handling backend to advertise the supported tags, allowing the program to present them in a drop-down menu so that the user can choose one of them. This could be done with a cross-file-format list of tag names, so that all backends respond with/to the same values. There's also the matter of getting/setting these tags, and that can be solved by making a single function like get_custom_tag/set_custom_tag, implemented in each file backend.

There's also the matter of handling those tags correctly when multiple files with incompatible file formats are selected, which may be a bit tricky to implement.

As for the UI aspect, the "Additional tags" could probably be moved into a listbox combobox. https://developer.gnome.org/hig/patterns/containers/boxed-lists.html

**Comment by Haaruun-I** [comment]
> The main issue with adding something like this is that every file tagging format has support for different tags. While there's overlap between most file formats, some more exotic tags may only be implemented in one format.

Most of those libraries would give an exception if you pass an unsupported tag correct? Those could be bubbled up and used to display some sort of error message. Or i can compile a list of the most common ones as a temporary fix. A better system could be done after #24, One backend would be allot easier to deal with than the three now.
